plasmapy/formulary/relativity.py

```diff
@@ -93,31 +94,57 @@ def Lorentz_factor(V: u.m / u.s):
 @validate_quantities(
     m={"can_be_negative": False}, validations_on_return={"can_be_negative": False}
 )
-def relativistic_energy(m: u.kg, v: u.m / u.s) -> u.Joule:
+@particle_input
+def relativistic_energy(
+    particle: ParticleLike,
+    V: u.m / u.s,
+    *,
+    mass_numb: Optional[Integral] = None,
+    Z: Optional[Integral] = None,
+) -> u.Joule:
     """
-    Calculate the relativistic energy (in joules) of an object of mass
-    ``m`` and velocity ``v``.
+    Calculate the sum of the mass energy and kinetic energy of a
+    relativistic body.
+
+    The total energy of a relativistic body is
 
     .. math::
 
-        E = γ m c^2
+        E = γ m c^2,
 
-    where :math:`γ` is the `Lorentz_factor`. This function returns the
-    sum of the mass energy and the kinetic energy.
+    where :math:`m` is the mass of the body and :math:`γ` is its
+    `~plasmapy.formulary.relativity.Lorentz_factor`.
 
     Parameters
     ----------
-    m : `~astropy.units.Quantity`
-        The mass in units convertible to kilograms.
+    particle : |particle-like|
+        A representation of a particle or a `~astropy.units.Quantity`
+        with units of mass.
```

Comment:
Still need to expand this.  We really need to standardize the parameter descriptions for particle-like inputs...